e2e-tests/playwright/specs/functional/system_console/permissions/team_access.spec.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {UserProfile} from '@mattermost/types/users';
+
+import {expect, PlaywrightExtended, test} from '@mattermost/playwright-lib';
+
+// setupSystemManagerRole configures the system manager with the given permission ("Can Edit", "Read only", "No access")
+// for the given section and subsection (e.g. "permission_section_reporting_site_statistics" and "permission_section_reporting_team_statistics").
+//
+// We do this via the system console and not the API because this page has multiple queries to build up the
+// final API call that are all ultimately part of the spec, and we want to test the effects of those, not
+// merely a hand-created version of the same.
+const setupDefaultSystemManagerRole = async (
+    pw: PlaywrightExtended,
+    adminUser: UserProfile,
+    sectionTestId: string,
+    subsectionTestId: string,
+    permissionText: string,
+) => {
+    // Login as admin and navigate to System Console
+    const {systemConsolePage: adminConsolePage} = await pw.testBrowser.login(adminUser);
+
+    // Go to System Console
+    await adminConsolePage.goto();
+
+    // Login to the System Console and navigate to Delegated Granular Administration
+    await adminConsolePage.sidebar.goToItem('Delegated Granular Administration');
+
+    // Find the System Manager row in the table
+    const systemManagerText = adminConsolePage.page.getByText('System Manager', {exact: true}).first();
```

Comment:
This looks good, no changes needed. Just sharing a quick thought: my usual cue for when something should be moved to a [page/component object model](https://github.com/mattermost/mattermost/tree/master/e2e-tests/playwright/lib/src/ui) is when I find myself frequently using `somePage.page.[locators]` to access elements. It's definitely an overhead to write from scratch, but it helps lay the foundation of our library.

Reply:
Thanks, @saturninoabril! I realized I was treading "new ground" here when I found no built-in support for this page. Propose the "inline approach" for now, and as we expand the test coverage, pulling it into that page model once we need to do it more than once. (This particular page definitely wasn't built for e2e manipulation :/)